controller/loginController.js

```diff
@@ -1,10 +1,51 @@
 const bcrypt = require('bcryptjs');
 const jwt = require('jsonwebtoken');
-let getUserCredentials = require('../model/getUserCredentials.js')
+let getUserCredentials = require('../model/getUserCredentials.js');
+let addMfaToken = require('../model/addMfaToken.js');
+const sgMail = require('@sendgrid/mail');
+
 
 const login = async (req, res) => {
     const { username, password } = req.body;
     
+    try {
+        if (!username || !password) {
+            return res.status(400).json({ error: 'Username and password are required' });
+         }
+        const user = await getUserCredentials(username, password);
+        if (user.length === 0) {
+            return res.status(401).json({ error: 'Invalid username or password' });
+        }
+
+        const isPasswordValid = await bcrypt.compare(password, user.password);
+        if (!isPasswordValid) {
+            return res.status(401).json({ error: 'Invalid username or password' });
+        }
+       
+
+        if (user.mfa_enabled) {
+            let token = Math.floor(100000 + Math.random() * 900000);
+
+            addMfaToken(user.user_id, token);
```

Comment:
Will need to await the addMfaToken function, otherwise the try block in login won't catch any errors in the async function and will happily send off an email even though it's failed

Reply:
Fixed Thanks